### PR TITLE
ci: Add test for minimum supported debug build of Python

### PR DESCRIPTION
**Pull request by Embers-of-the-Fire**
As per comment in #5847.

> Makes sense. We should probably also add a `Py_REF_DEBUG` build to CI on the lowest supported Python version.

*Note* that the original debug build downloads prebuilt standalone binaries from `python-build-standalone`, but the prebuilt binaries are limited (for old versions). In the latest release there's only 3.10.19 available, and I could not find a 3.7 build in all releases.
This PR pins it to `3.10.19-3.14.3` from `20260211` release.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
>  In the latest release there's only 3.10.19 available

That's fine, 3.10 will be the minimum supported version soon enough. We're already way overdue for dropping 3.7 and 3.8 support.

**Comment by ngoldbaum** [comment]
I'll defer to @davidhewitt on whether it's OK to add another CI job to the full CI matrix. I added a ci-build-full label to make the new job actually run in this PR. You'll need to push another commit to trigger the tests again.

**Comment by ngoldbaum** [comment]
The docs build error should go away once #5854 goes in.

**Comment by Embers-of-the-Fire** [comment]
The error seems to be irrelavent with the CI :(

**Comment by davidhewitt** [comment]
I think #5859 will fix the CI issues.

I am also looking at #5447 which should make this PR simpler to land & maintain.

**Comment by Embers-of-the-Fire** [comment]
> I am also looking at #5447 which should make this PR simpler to land & maintain.

:) agreed. Relying on *curl*s is less than ideal. I'll have an eye on that.

**Comment by davidhewitt** [comment]
@Embers-of-the-Fire I think this should be ready to update against current main. Thanks for the patience!

**Comment by Embers-of-the-Fire** [comment]
@davidhewitt Updated. But the error shown in https://github.com/PyO3/pyo3/actions/runs/27054706739/job/79856801452?pr=5852 seems to be irrelavent with the CI change.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -603,6 +603,24 @@ jobs:
...
+      - uses: actions/checkout@v6.0.3
+      - uses: Swatinem/rust-cache@v2
+        with:
+          save-if: ${{ github.ref == 'refs/heads/main' || contains(github.event.pull_request.labels.*.name, 'CI-save-pr-cache') }}
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rust-src
+      - uses: astral-sh/setup-uv@v7
+        with:
+          python-version: "3.10+debug"
```
Comment (davidhewitt) [inline .github/workflows/ci.yml]: Testing locally I have some issues with 3.10+debug finding the correct environment to start up when run from uv in the PyO3 test suite (similar to https://github.com/PyO3/pyo3/issues/4813#issuecomment-2616922117), so let's bump this to 3.11 which seems to work locally for me

```suggestion
          python-version: "3.11+debug"
```
